dockers/docker-orchagent/orchagent.sh

```diff
@@ -73,9 +73,14 @@ else
 fi
 
 # Enable ZMQ for SmartSwitch
-LOCALHOST_SUBTYPE=`sonic-db-cli CONFIG_DB hget localhost "subtype"`
+LOCALHOST_SUBTYPE=`sonic-db-cli CONFIG_DB hget "DEVICE_METADATA|localhost" "subtype"`
 if [[ x"${LOCALHOST_SUBTYPE}" == x"SmartSwitch" ]]; then
-    ORCHAGENT_ARGS+=" -q tcp://127.0.0.1:8100"
+    mgmt_ip=$( ip -4 -o addr show eth0 | awk '{print $4}' | cut -d'/' -f1 )
```

Comment:
According to the HLD, It's not eth0 but eth0-midplane.https://github.com/Pterosaur/SONiC/blob/11b0514903a7cead54a6fbd30b1f0d527413e73e/doc/smart-switch/ip-address-assigment/smart-switch-ip-address-assignment.md#dpu

Reply:
Fixed
I check the starlab dpu, there is no eth0-midplane, so the code will try eth0-midplane first